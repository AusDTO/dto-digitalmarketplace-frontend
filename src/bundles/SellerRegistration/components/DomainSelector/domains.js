export default [
    {
        key: 'deliverygov',
        label: 'Agile delivery and Governance',
        description: 'Agile delivery and governance covers all services relating to the delivery and management of a digital service or product. Examples include business analysis, delivery management, performance and web analysis, and product management.'
    },
    {
        key: 'changetransformation',
        label: 'Change and Transformation',
        description: 'Change and transformation is about embedding more efficient work culture and practices in government. Examples include digital transformation advice, change management and transitioning to new ways of working.'
    },
    {
        key: 'contentpub',
        label: 'Content and Publishing',
        description: 'Content and publishing covers all services that provide or manage of any type of online or offline content. Examples include content design, writing, translation, photography, video, interactive, illustration and animation.',
    },
    {
        key: 'cyber',
        label: 'Cyber security',
        description: 'Cyber security protects the Australian government and its citizensâ€™ interests online. Examples include cyber security advice, cyber security education, information security, assessments, ethical hacking, monitoring, incident response, data privacy and digital forensics.'
    },
    {
        key: 'datasci',
        label: 'Data science',
        description: 'Data science brings analytical rigour and statistical methods to measure quality, improve services and model behaviour. Examples include statistical and scenario modelling and data visualisation.'
    },
    {
        key: 'ictprocurement',
        label: 'Digital sourcing and ICT procurement',
        description: 'Digital sourcing includes the identification and application of activities surrounding ICT procurement. Examples include providing subject matter expertise and recommendations, managing complex or high risk procurements, probity, and managing internal documentation.'
    },
    {
        key: 'emergtech',
        label: 'Emerging technologies',
        description: 'Emerging technology is about developing proof of concepts using recent digital advances for commercial or government purposes. Examples include artificial intelligence, robotics, wearables, Internet of Things and augmented or virtual reality.'
    },
    {
        key: 'ictservices',
        label: 'ICT managed services',
        description: 'ICT managed services is the practice of outsourcing certain processes and functions to improve operations and cut expenses. Examples include end user services, network management services, management of cloud implementations, service desk operations and onside support services.'
    },
    {
        key: 'ictaudit',
        label: 'ICT risk management and audit activities',
        description: 'ICT risk management and audit involves the identification, evaluation, and prioritization of risks and the activities to minimize, monitor, and control the impact of unfortunate events. Examples include saving initiatives, reporting obligations, investment framework, business case development and costing frameworks.'
    },
    {
        key: 'marketingcomms',
        label: 'Marketing, Communications and Engagement',
        description: 'Marketing, communications and engagement create awareness, adoption and ongoing use of a digital service. Examples include advertising, PR, event marketing, content marketing, social, SEO and growth hacking.',
    },
    {
        key: 'engineering',
        label: 'Software engineering and Development',
        description: 'Engineering and development cover the technical aspects of the creation of a digital service. Examples include web development, devops, technical leadership, architecture, continuous delivery, testing and quality assurance.',
    },
    {
        key: 'strategy',
        label: 'Strategy and Policy',
        description: 'Strategy and policy relates to the higher level planning and policy considerations of a digital product or service. Examples include business case development, policy analysis and policy advice.',
    },
    {
        key: 'ops',
        label: 'Support and Operations',
        description: 'Support and operations covers the day-to-day activities that can be needed when running a digital service. Examples include administration, business operations, data entry, call centres and tech support.',
    },
    {
        key: 'training',
        label: 'Training, Learning and Development',
        description: 'Training, learning and development builds skills and expertise to deliver government services digitally. Examples include training needs analysis, developing and delivering learning solutions, coaching and mentoring and learning impact evaluation.'
    },
    {
        key: 'userresearch',
        label: 'User research and Design',
        description: 'User research and design covers all activities relating to the research and design of a service. Examples include user research, interaction, inclusive, and service design, participant recruitment and facilities for usability testing.'
    }
];
