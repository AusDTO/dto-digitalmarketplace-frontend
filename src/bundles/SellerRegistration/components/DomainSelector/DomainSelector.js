import React from 'react';
import { bindActionCreators } from 'redux'
import {connect} from 'react-redux';
import {Form, Control} from 'react-redux-form';
import kebabCase from 'lodash/kebabCase';
import isNumber from 'lodash/isNumber';
import { Link } from 'react-router-dom';

import Layout         from '../../../../shared/Layout';
import BaseForm       from '../../../../shared/form/BaseForm';
import ErrorBox       from '../../../../shared/form/ErrorBox';
import StatefulError  from '../../../../shared/form/StatefulError';
import formProps      from '../../../../shared/reduxModules/formPropsSelector';
import StepNav        from '../StepNav';

import { actions }    from '../../redux/modules/application';
import { actions as stepActions }    from '../../redux/modules/steps';

import domains from './domains';

class DomainSelector extends BaseForm {

  static propTypes = {
    recruiter: React.PropTypes.string
  }

    render() {
        const {model, supplierCode, action, csrf_token, buttonText, children, actions, onSubmit, recruiter, nextRoute} = this.props;
        let header = (
            <header>
                <h1 tabIndex="-1">What services will you offer?</h1>
                <p> If you only have software products to offer, <Link
                    to="/products"
                    onClick={() => {
                        // Hardcoded is ugly.
                        actions.stepComplete('casestudy');
                        actions.stepComplete('digital');
                        actions.navigateToStep('/products');
                    }}>skip to the product section.
                </Link></p>
                <p>Select the services you have the expertise to provide.  For each you choose you will need to provide
                    a case study and referee to ensure you meet our <a href="/assessment-criteria" target="_blank" rel="external">assessment criteria</a></p>

            </header>
        )

        if (isNumber(supplierCode)) {
            header = (
                <header>
                    <h1 tabIndex="-1">What services will you offer?</h1>
                    <p>The services you are currently approved for are already ticked. Select any others you have the skills and experience to provide.
                        You will need to provide a case study and referee for each to meet the <a href="/assessment-criteria"
                                                                                          target="_blank" rel="external">assessment criteria</a>
                    </p>
                </header>
            )
        }

        if (recruiter === 'yes') {
            header = (
                <header>
                    <h1 tabIndex="-1">What services do your candidates offer?</h1>
                </header>
            )
        }

        return (
            <Layout>
                {header}
                <article role="main">

                    <ErrorBox focusOnMount={true} model={model}/>

                    {/* This error will never actually render */}
                    <StatefulError
                        model={`${model}.services`}
                        id="services"
                        messages={{
                            services: 'You must select at least one service from the services below.'
                        }}
                    />

                    <Form
                        model={model}
                        action={action}
                        method="post"
                        id="DomainSelector__create"
                        validators={{
                            services: (services) => {
                                return Object
                                    .keys(services)
                                    .filter(s => services[s])
                                    .length;
                            }
                        }}
                        onSubmit={onSubmit}>

                        {csrf_token && (
                            <input type="hidden" name="csrf_token" id="csrf_token" value={csrf_token}/>
                        )}

                        <fieldset className="field">
                            {domains.map(({key, label, description}, i) => {
                                return (
                                    <div key={key}>
                                        <Control.checkbox
                                            model={`${model}.services.${label}`}
                                            id={kebabCase(label)}
                                            name={kebabCase(label)}
                                            value={label}
                                        />
                                        <label htmlFor={kebabCase(label)}>{label} <p>{description}</p></label>
                                    </div>
                                )
                            })}
                        </fieldset>

                        {children}

                        <StepNav buttonText={buttonText} to={nextRoute}/>
                    </Form>
                </article>
            </Layout>
        )
    }
}

DomainSelector.defaultProps = {
    buttonText: 'Save and continue',
    title: 'What services will you offer? ',
    recruiter: ''
}

const mapStateToProps = (state) => {
    return {
        supplierCode: state.application.supplier_code,
        ...formProps(state, 'domainSelectorForm')
    }
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators({ ...actions, ...stepActions }, dispatch),
    dispatch
})

export default connect(mapStateToProps, mapDispatchToProps)(DomainSelector);