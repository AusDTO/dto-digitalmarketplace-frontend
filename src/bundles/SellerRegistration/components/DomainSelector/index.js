import DomainSelector from './DomainSelector';

export default DomainSelector;