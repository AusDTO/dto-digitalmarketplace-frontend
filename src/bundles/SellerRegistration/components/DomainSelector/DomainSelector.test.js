// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store';
import { StaticRouter } from 'react-router-dom';

import DomainSelector from './DomainSelector'
import createStore from '../../redux/create-signup'

test('DomainSelector renders', () => {
  const store = createStore({
    form_options: {
      csrf_token: 'abc'
    },
    options: {
      serverRender: false
    }
  });

  const component = renderer.create(
    <StaticRouter context={{}}>
      <Provider store={store}>
        <DomainSelector />
      </Provider>
    </StaticRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('DomainSelector renders with populated fields', () => {
  const store = createStore({
    form_options: {
      csrf_token: 'abc'
    },
    domainSelectorForm: {
      services: {
        'User research': true,
        'Development': true,
        'Content management': false
      }
    },
    options: {
      serverRender: false
    }
  });

  const component = renderer.create(
    <StaticRouter context={{}}>
      <Provider store={store}>
        <DomainSelector />
      </Provider>
    </StaticRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

