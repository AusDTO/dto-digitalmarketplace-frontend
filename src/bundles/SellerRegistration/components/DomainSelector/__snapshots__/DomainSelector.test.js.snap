// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSelector renders 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      What services will you offer?
    </h1>
    <p>
       If you only have software products to offer, 
      <a
        href="/products"
        onClick={[Function]}
      >
        skip to the product section.
      </a>
    </p>
    <p>
      Select the services you have the expertise to provide.  For each you choose you will need to provide a case study and referee to ensure you meet our 
      <a
        href="/assessment-criteria"
        rel="external"
        target="_blank"
      >
        assessment criteria
      </a>
    </p>
  </header>
  <article
    role="main"
  >
    <form
      action={undefined}
      id="DomainSelector__create"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <input
        id="csrf_token"
        name="csrf_token"
        type="hidden"
        value="abc"
      />
      <fieldset
        className="field"
      >
        <div>
          <input
            checked={false}
            disabled={false}
            id="strategy-and-policy"
            name="strategy-and-policy"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Strategy and Policy"
          />
          <label
            htmlFor="strategy-and-policy"
          >
            Strategy and Policy
             
            <p>
              Strategy and policy relates to the higher level planning and policy considerations of a digital product or service. Examples include business case development, policy analysis and policy advice.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="user-research-and-design"
            name="user-research-and-design"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="User research and Design"
          />
          <label
            htmlFor="user-research-and-design"
          >
            User research and Design
             
            <p>
              User research and design covers all activities relating to the research and design of a service. Examples include user research, interaction, inclusive, and service design, participant recruitment and facilities for usability testing.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="agile-delivery-and-governance"
            name="agile-delivery-and-governance"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Agile delivery and Governance"
          />
          <label
            htmlFor="agile-delivery-and-governance"
          >
            Agile delivery and Governance
             
            <p>
              Agile delivery and governance covers all services relating to the delivery and management of a digital service or product. Examples include business analysis, delivery management, performance and web analysis, and product management.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="software-engineering-and-development"
            name="software-engineering-and-development"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Software engineering and Development"
          />
          <label
            htmlFor="software-engineering-and-development"
          >
            Software engineering and Development
             
            <p>
              Engineering and development cover the technical aspects of the creation of a digital service. Examples include web development, devops, technical leadership, architecture, continuous delivery, testing and quality assurance.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="support-and-operations"
            name="support-and-operations"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Support and Operations"
          />
          <label
            htmlFor="support-and-operations"
          >
            Support and Operations
             
            <p>
              Support and operations covers the day-to-day activities that can be needed when running a digital service. Examples include administration, business operations, data entry, call centres and tech support.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="content-and-publishing"
            name="content-and-publishing"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Content and Publishing"
          />
          <label
            htmlFor="content-and-publishing"
          >
            Content and Publishing
             
            <p>
              Content and publishing covers all services that provide or manage of any type of online or offline content. Examples include content design, writing, translation, photography, video, interactive, illustration and animation.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="change-training-and-transformation"
            name="change-training-and-transformation"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Change, Training and Transformation"
          />
          <label
            htmlFor="change-training-and-transformation"
          >
            Change, Training and Transformation
             
            <p>
              Change, training and transformation is about embedding more efficient work culture and practices in government. Examples include digital transformation advice, agile coaching, change management and training in new ways of working.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="marketing-communications-and-engagement"
            name="marketing-communications-and-engagement"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Marketing, Communications and Engagement"
          />
          <label
            htmlFor="marketing-communications-and-engagement"
          >
            Marketing, Communications and Engagement
             
            <p>
              Marketing, communications and engagement create awareness, adoption and ongoing use of a digital service. Examples include advertising, PR, event marketing, content marketing, social, SEO and growth hacking.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="cyber-security"
            name="cyber-security"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Cyber security"
          />
          <label
            htmlFor="cyber-security"
          >
            Cyber security
             
            <p>
              Cyber security protects the Australian government and its citizensâ€™ interests online. Examples include cyber security advice, cyber security education, information security, assessments, ethical hacking, monitoring, incident response, data privacy and digital forensics.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="data-science"
            name="data-science"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Data science"
          />
          <label
            htmlFor="data-science"
          >
            Data science
             
            <p>
              Data science brings analytical rigour and statistical methods to measure quality, improve services and model behaviour. Examples include statistical and scenario modelling and data visualisation.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="emerging-technologies"
            name="emerging-technologies"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Emerging technologies"
          />
          <label
            htmlFor="emerging-technologies"
          >
            Emerging technologies
             
            <p>
              Emerging technology is about developing proof of concepts using recent digital advances for commercial or government purposes. Examples include artificial intelligence, robotics, wearables, Internet of Things and augmented or virtual reality.
            </p>
          </label>
        </div>
      </fieldset>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Save and continue
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;

exports[`DomainSelector renders with populated fields 1`] = `
<div
  className="layout"
>
  <header>
    <h1
      tabIndex="-1"
    >
      What services will you offer?
    </h1>
    <p>
       If you only have software products to offer, 
      <a
        href="/products"
        onClick={[Function]}
      >
        skip to the product section.
      </a>
    </p>
    <p>
      Select the services you have the expertise to provide.  For each you choose you will need to provide a case study and referee to ensure you meet our 
      <a
        href="/assessment-criteria"
        rel="external"
        target="_blank"
      >
        assessment criteria
      </a>
    </p>
  </header>
  <article
    role="main"
  >
    <form
      action={undefined}
      id="DomainSelector__create"
      method="post"
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <input
        id="csrf_token"
        name="csrf_token"
        type="hidden"
        value="abc"
      />
      <fieldset
        className="field"
      >
        <div>
          <input
            checked={false}
            disabled={false}
            id="strategy-and-policy"
            name="strategy-and-policy"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Strategy and Policy"
          />
          <label
            htmlFor="strategy-and-policy"
          >
            Strategy and Policy
             
            <p>
              Strategy and policy relates to the higher level planning and policy considerations of a digital product or service. Examples include business case development, policy analysis and policy advice.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="user-research-and-design"
            name="user-research-and-design"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="User research and Design"
          />
          <label
            htmlFor="user-research-and-design"
          >
            User research and Design
             
            <p>
              User research and design covers all activities relating to the research and design of a service. Examples include user research, interaction, inclusive, and service design, participant recruitment and facilities for usability testing.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="agile-delivery-and-governance"
            name="agile-delivery-and-governance"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Agile delivery and Governance"
          />
          <label
            htmlFor="agile-delivery-and-governance"
          >
            Agile delivery and Governance
             
            <p>
              Agile delivery and governance covers all services relating to the delivery and management of a digital service or product. Examples include business analysis, delivery management, performance and web analysis, and product management.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="software-engineering-and-development"
            name="software-engineering-and-development"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Software engineering and Development"
          />
          <label
            htmlFor="software-engineering-and-development"
          >
            Software engineering and Development
             
            <p>
              Engineering and development cover the technical aspects of the creation of a digital service. Examples include web development, devops, technical leadership, architecture, continuous delivery, testing and quality assurance.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="support-and-operations"
            name="support-and-operations"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Support and Operations"
          />
          <label
            htmlFor="support-and-operations"
          >
            Support and Operations
             
            <p>
              Support and operations covers the day-to-day activities that can be needed when running a digital service. Examples include administration, business operations, data entry, call centres and tech support.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="content-and-publishing"
            name="content-and-publishing"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Content and Publishing"
          />
          <label
            htmlFor="content-and-publishing"
          >
            Content and Publishing
             
            <p>
              Content and publishing covers all services that provide or manage of any type of online or offline content. Examples include content design, writing, translation, photography, video, interactive, illustration and animation.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="change-training-and-transformation"
            name="change-training-and-transformation"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Change, Training and Transformation"
          />
          <label
            htmlFor="change-training-and-transformation"
          >
            Change, Training and Transformation
             
            <p>
              Change, training and transformation is about embedding more efficient work culture and practices in government. Examples include digital transformation advice, agile coaching, change management and training in new ways of working.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="marketing-communications-and-engagement"
            name="marketing-communications-and-engagement"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Marketing, Communications and Engagement"
          />
          <label
            htmlFor="marketing-communications-and-engagement"
          >
            Marketing, Communications and Engagement
             
            <p>
              Marketing, communications and engagement create awareness, adoption and ongoing use of a digital service. Examples include advertising, PR, event marketing, content marketing, social, SEO and growth hacking.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="cyber-security"
            name="cyber-security"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Cyber security"
          />
          <label
            htmlFor="cyber-security"
          >
            Cyber security
             
            <p>
              Cyber security protects the Australian government and its citizensâ€™ interests online. Examples include cyber security advice, cyber security education, information security, assessments, ethical hacking, monitoring, incident response, data privacy and digital forensics.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="data-science"
            name="data-science"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Data science"
          />
          <label
            htmlFor="data-science"
          >
            Data science
             
            <p>
              Data science brings analytical rigour and statistical methods to measure quality, improve services and model behaviour. Examples include statistical and scenario modelling and data visualisation.
            </p>
          </label>
        </div>
        <div>
          <input
            checked={false}
            disabled={false}
            id="emerging-technologies"
            name="emerging-technologies"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            type="checkbox"
            value="Emerging technologies"
          />
          <label
            htmlFor="emerging-technologies"
          >
            Emerging technologies
             
            <p>
              Emerging technology is about developing proof of concepts using recent digital advances for commercial or government purposes. Examples include artificial intelligence, robotics, wearables, Internet of Things and augmented or virtual reality.
            </p>
          </label>
        </div>
      </fieldset>
      <div
        styleName="nav-row"
      >
        <button
          styleName="nav-button"
          type="submit"
        >
          Save and continue
        </button>
        <button
          className="save-button"
          onClick={[Function]}
          styleName="nav-button"
        >
          Save and finish later
        </button>
        <button
          className="button-secondary"
          onClick={[Function]}
          styleName="nav-link skip-button"
        >
          Skip for now
        </button>
      </div>
    </form>
  </article>
</div>
`;
